src/api.ts

```diff
@@ -522,56 +577,176 @@ function createApi({
+      ).run({
+        id: styleId,
+        stylejson: JSON.stringify(styleToSave),
+        etag,
+        upstreamUrl,
+        sourceIdToTilesetId: JSON.stringify(sourceIdToTilesetId),
+      })
 
-      return addOfflineUrls(offlineStyle)
+      return {
+        style: addOfflineUrls({
+          sourceIdToTilesetId,
+          style: styleToSave,
+          styleId,
+        }),
+        id: styleId,
+      }
     },
 
-    async putStyle(id, style) {
-      db.prepare(
-        'INSERT INTO Style (id, stylejson) VALUES (:id, :stylejson)'
-      ).run({ id, stylejson: style })
-      return { ...style, id }
+    // TODO: May need to accept an access token
+    async updateStyle(id, style) {
+      if (!styleExists(id)) {
+        throw new NotFoundError(id)
+      }
+
+      const { sources, sourceIdToTilesetId } = await createOfflineSources({
+        sources: style.sources,
+      })
+
+      const styleToSave: StyleJSON = {
+        ...(await uncompositeStyle({
+          ...style,
+          sources,
+        })),
+      }
+
+      db.prepare<{
+        id: string
+        sourceIdToTilesetId: string
+        stylejson: string
+      }>(
+        'UPDATE Style SET stylejson = :stylejson, sourceIdToTilesetId = :sourceIdToTilesetId WHERE id = :id'
+      ).run({
+        id,
+        sourceIdToTilesetId: JSON.stringify(sourceIdToTilesetId),
+        stylejson: JSON.stringify(styleToSave),
+      })
+
+      return addOfflineUrls({
+        sourceIdToTilesetId,
+        style: styleToSave,
+        styleId: id,
+      })
     },
 
-    async listStyles(limit?: number) {
-      const baseQuery = 'SELECT stylejson FROM Style'
-      const stmt =
-        limit !== undefined
-          ? db.prepare(`${baseQuery} LIMIT ?`).bind(limit)
-          : db.prepare(baseQuery)
+    async listStyles() {
+      const styles: ({ name?: string; style: StyleJSON } & IdResource)[] = []
 
-      return stmt
+      db.prepare('SELECT id, stylejson, sourceIdToTilesetId FROM Style')
         .all()
-        .map((row: { stylejson: string }) => JSON.parse(row.stylejson))
+        .forEach(
+          (row: {
+            id: string
+            stylejson: string
+            sourceIdToTilesetId: string
+          }) => {
+            let style: StyleJSON
+            let sourceIdToTilesetId: SourceIdToTilesetId
+
+            try {
+              style = JSON.parse(row.stylejson)
+              sourceIdToTilesetId = JSON.parse(row.sourceIdToTilesetId)
+            } catch (err) {
+              throw new ParseError(err)
+            }
+
+            const adjustedStyle = addOfflineUrls({
+              style,
+              styleId: row.id,
+              sourceIdToTilesetId,
+            })
+
+            // TODO:
+            // - Should we have a fallback name here or let client handle?
+            // - Providing the whole stylejson is temporary and won't be needed once static map generation is implemented
```

Comment:
I guess that's the question we were trying to address. Thought it would make sense to just provide the whole style so that the client could render the maps immediately after getting the response, as opposed to having to make subsequent requests for each style after the initial response. not super opinionated on it since this will change once thumbnail generation is implemented

Reply:
Addressed via https://github.com/digidem/mapeo-mapserver/pull/13/commits/a3983164215ba1ee61a314cc4e447f527ca36428